src/dialect/mod.rs

```diff
@@ -490,6 +490,26 @@ pub trait Dialect: Debug + Any {
         false
     }
 
+    /// Return true if the dialect supports creating enum types
+    ///
+    /// Example:
+    /// ```sql
+    /// CREATE TYPE some_enum AS ENUM('one', 'two', 'three');
+    /// ```
+    fn supports_create_type_as_enum(&self) -> bool {
+        false
+    }
+
+    /// Return true if the dialect supports ALTER TYPE statements
+    ///
+    /// Example:
+    /// ```sql
+    /// ALTER TYPE some_enum ADD VALUE 'some_value';
+    /// ```
+    fn supports_alter_type(&self) -> bool {
+        false
+    }
```

Comment:
Thinking we could skip the dialect flags for this case and have the parse the statement unconditionally, if it does not conflict with existing statements?